crypto/crypto.go

```diff
@@ -119,8 +93,15 @@ type InMemorySigner struct {
 var _ Signer = (*InMemorySigner)(nil)
 
 // NewInMemorySigner initializes and returns a new in-memory signer with the provided private key
-// and hasher.
+// and hashing algorithm.
 func NewInMemorySigner(privateKey PrivateKey, hashAlgo HashAlgorithm) InMemorySigner {
+	// check compatibility to form a signing key
+	if !CompatibleAlgorithms(privateKey.Algorithm(), hashAlgo) {
+		// TODO: panic?
+		return InMemorySigner{}
```

Comment:
Should an error be returned here as well?

Reply:
Good catch, I missed getting back to that. 
If this was an internal function I would add an error. For the public function, this will cause a breaking change. If you're okay with that I will add it. Any preference?

Reply:
I would say let us break it while we are not 1.0 after that it will be just harder. The problem if we don't do that we will just support "anti-patterns" and that's not good either.

Reply:
https://github.com/onflow/flow-go-sdk/pull/283/commits/7c53dc1d5babe18187ec5c1b63a34c31397df3dc